pkg/utils/net-attach-def.go

```diff
@@ -122,6 +122,67 @@ func GetNetworkStatus(pod *corev1.Pod) ([]v1.NetworkStatus, error) {
 	return netStatuses, err
 }
 
+// CreateNetworkStatuses creates an array of NetworkStatus from CNI result
+// Not to be confused with CreateNetworkStatus (singular)
+// This is the preferred method and picks up when CNI ADD results contain multiple container interfaces
+func CreateNetworkStatuses(r cnitypes.Result, networkName string, defaultNetwork bool, dev *v1.DeviceInfo) ([]*v1.NetworkStatus, error) {
+	var networkStatuses []*v1.NetworkStatus
+	// indexMap is from original CNI result index to networkStatuses index
+	indexMap := make(map[int]int)
+
+	// Convert whatever the IPAM result was into the current Result type
+	result, err := cni100.NewResultFromResult(r)
+	if err != nil {
+		return nil, fmt.Errorf("error converting the type.Result to cni100.Result: %v", err)
+	}
+
+	// Discover default routes upfront and reuse them if necessary.
+	var usegateway []string
+	for _, route := range result.Routes {
+		if isDefaultRoute(route) {
+			usegateway = append(usegateway, route.GW.String())
+		}
+	}
+
+	// Same for DNS
+	v1dns := convertDNS(result.DNS)
+
+	// Initialize NetworkStatus for each container interface (e.g. with sandbox present)
+	keepidx := 0
+	for i, iface := range result.Interfaces {
+		if iface.Sandbox != "" {
+			ns := &v1.NetworkStatus{
+				Name:       networkName,
+				Default:    defaultNetwork,
+				Interface:  iface.Name,
+				Mac:        iface.Mac,
+				Mtu:        iface.Mtu,
+				IPs:        []string{},
+				Gateway:    usegateway,
+				DeviceInfo: dev,
+				DNS:        *v1dns,
+			}
+			networkStatuses = append(networkStatuses, ns)
+			// Map original index to the new slice index
+			indexMap[i] = keepidx
+			keepidx++
+		}
+	}
+
+	// Map IPs to network interface based on index
+	for _, ipconfig := range result.IPs {
```

Comment:
nit: let's be consistent and use camel case everywhere

```suggestion
	for _, ipConfig := range result.IPs {
```